### PR TITLE
New Tricks

### PR DESCRIPTION
**Pull request by debriefff**

### CODE REVIEW

File: common_seq_method.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+
+class Foo():
+    def bar(self, *args, **kwargs):
+        print('method bar works')
+
+
+sequence = [Foo() for i in range(5)]
+
+# in python3 map returns iterator so we must ask python ro process elements by list()
```
Comment (brennerm) [inline common_seq_method.py]: change "ro" to "to" ;)

File: unique_by_attr.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    If we have some sequence of objects and want to remove items with the same attribute value
+    Python creates a dict, where keys are value if attribute (bar in our case), values are object of the sequence.
+    After that the dict is transformed back to list
+
+    Note: in result we save the last from repeating elements (item2 in our case)!
+"""
+
+
+class Foo():
+    pass
```
Comment (brennerm) [inline unique_by_attr.py]: Could you change "bar" to be a class attribute and set it's value with the constructor?

**Comment by brennerm** [comment]
Pls have a look at my notes :)

File: unique_by_attr.py (hunk 2 above)
Comment (debriefff) [inline unique_by_attr.py]: I can, but the main idea of "bar" attribute is to show  which elements python will remove from secuence and why. So I suppose setting that attr for each item individually suits here pretty nice

Comment (brennerm) [inline unique_by_attr.py]: You can set them individually like that:

``` python
class Foo:
    def __init__(self, value):
        self.bar = value

item1 = Foo(15)
item2 = Foo(15)
item3 = Foo(5)
```

Comment (debriefff) [inline unique_by_attr.py]: m, ok

**Comment by brennerm** [comment]
Thank you! :)

**Pull request merged**